src/controllers/csv-processor.ts

```diff
@@ -316,18 +355,82 @@ export const processCSVFromBlobStorage = async (
         logger.error(err);
         return {
             success: false,
+            status: 500,
             errors: [
                 {
                     field: 'csv',
                     message: [
                         { lang: ENGLISH, message: t('errors.download_from_blobstorage', { lng: ENGLISH }) },
                         { lang: WELSH, message: t('errors.download_from_blobstorage', { lng: WELSH }) }
                     ],
-                    tag: { name: 'errors.download_from_datalake', params: {} }
+                    tag: { name: 'errors.download_from_blobstorage', params: {} }
                 }
             ],
             dataset_id: dataset.id
         };
     }
     return processCSVData(buff, page, size, dataset, importObj);
 };
+
+export const moveFileToDataLake = async (importObj: FileImport) => {
+    const blobStorageService = new BlobStorageService();
+    const datalakeService = new DataLakeService();
+    try {
+        const fileStream = await blobStorageService.getReadableStream(importObj.filename);
+        await datalakeService.uploadFileStream(importObj.filename, fileStream);
+        await blobStorageService.deleteFile(importObj.filename);
+    } catch (err) {
+        logger.error(err);
+        throw new Error('Error moving file to datalake');
+    }
+};
+
+export const removeTempfileFromBlobStorage = async (importObj: FileImport) => {
```

Comment:
No... Because the moveFile only handles copying the file from BlobStorage to the Datalake... Not the remove operation.  I kept them separate as they can both fail and its good to have an over view of why it failed.  Technically removing the file probably isn't fatal to the operation of the system.  So we might choose to not return an error if we fail to remove the file.

Reply:
Yes... If the user backs out of the dataset upload... Decides they've uploaded the wrong file this cleans up that file.  See the `router.delete('/:dataset_id/revision/by-id/:revision_id/import/by-id/:import_id')` route which is part of the transition from previewing the datafile to displaying the sources or choosing to upload a different file.